src/ceph-disk/ceph_disk/main.py

```diff
@@ -1666,26 +1666,33 @@ def create_partition(self, uuid, name, size=0, num=0):
         if num == 0:
             num = get_free_partition_index(dev=self.path)
         if size > 0:
-            new = '--new={num}:0:+{size}M'.format(num=num, size=size)
+            end = '{size}M'.format(size=size)
             if size > self.get_dev_size():
                 LOG.error('refusing to create %s on %s' % (name, self.path))
                 LOG.error('%s size (%sM) is bigger than device (%sM)'
                           % (name, size, self.get_dev_size()))
                 raise Error('%s device size (%sM) is not big enough for %s'
                             % (self.path, self.get_dev_size(), name))
         else:
-            new = '--largest-new={num}'.format(num=num)
+            end = '100%'
 
         LOG.debug('Creating %s partition num %d size %d on %s',
                   name, num, size, self.path)
         command_check_call(
             [
+               'parted',
```

Comment:
It might be misguided/misinformed, but I was under the impression that sgdisk had some problems and that parted was a newer, better supported tool.  Given we're moving toward ceph-volume anyway it might not be worth the trouble to make a conversion.